browser/ui/webui/brave_new_tab/custom_image_chooser.cc

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/webui/brave_new_tab/custom_image_chooser.h"
+
+#include <utility>
+
+#include "brave/components/l10n/common/localization_util.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/chrome_select_file_policy.h"
+#include "chrome/grit/generated_resources.h"
+#include "content/public/browser/web_contents.h"
+#include "ui/shell_dialogs/selected_file_info.h"
+
+namespace brave_new_tab {
+
+CustomImageChooser::CustomImageChooser(content::WebContents& web_contents)
```

Comment:
why aren't we just using the existing functionality in `BraveNewTabPageHandler`?